### PR TITLE
Use cursor signaling

### PR DESCRIPTION
**Pull request by ecarreras**
Now is possible to enqueue jobs when the transaction is commited, using a new parameter on `job` definition `on_commit=True`. e.g.:

```python
@job(queue='high', on_commit=True)
def hard_job_in_the_night(cursor, uid, ids, context=None):
    ...
```

Default behavior is `on_commit=False` to maintain backward compatibility

Needs https://github.com/gisce/erp/pull/6412

As commented in https://github.com/gisce/oorq/pull/74#issuecomment-383154672 we have created and extension of `AutoWorker` to work with the new cursor signaling system.

:warning: **NOTE**: If some jobs are executed in AutoWorker and the job is configured `on_commit` we should use our version of AutoWorker

```python
from autoworker import AutoWorker
...
# And then
aw = AutoWorker(queue='make_invoices', default_result_ttl=24 * 3600)
aw.work()
```

to:

```python
from oorq.autoworker import AutoWorker
...
# And then
aw = AutoWorker(queue='make_invoices', default_result_ttl=24 * 3600)
aw.work(cursor)
```

:information_source: cursor is the cursor we want to bind the start of autoworkers

/cc @rchamorro

### CODE REVIEW

**Comment by ecarreras** [comment]
Check autoworkers

**Comment by ecarreras** [comment]
We have to inherit autoworkers and modify `work` method to register in memory and then when the transaction is committed the call the real `work`.

Then in all the modules that are using autoworkers we need to use the autoworker from oorq module

/cc @tinogis @guilleJB

File: oorq/decorators.py [hunk 1]
```diff
@@ -58,17 +96,22 @@ def f_job(*args, **kwargs):
...
                 job_kwargs = kwargs
-                job = q.enqueue(
+                job = Job.create(
                     execute,
-                    depends_on=current_job,
-                    result_ttl=self.result_ttl,
                     args=job_args,
                     kwargs=job_kwargs,
+                    result_ttl=self.result_ttl,
+                    depends_on=current_job
```
Comment (guilleJB) [inline oorq/decorators.py]:
```suggestion
                    depends_on=current_job,
```
Fix missing coma

**Review by guilleJB — CHANGES_REQUESTED** [review]
Little typo

**Comment by ecarreras** [comment]
It seems exists a bug in RQ when using `enqueue_job()` dependencies aren't working...

**Comment by ecarreras** [comment]
> It seems exists a bug in RQ when using `enqueue_job()` dependencies aren't working...

Tested with `test_oorq` module and method `test_dependency_job_on_commit()` and it works as expected enqueuing the job when the first job is finished

**Pull request merged**